src/libraries/System.IO.FileSystem.Watcher/tests/Args.FileSystemEventArgs.cs

```diff
@@ -8,28 +8,61 @@ namespace System.IO.Tests
     public class FileSystemEventArgsTests
     {
         [Theory]
-        [InlineData(WatcherChangeTypes.Changed, "C:", "foo.txt")]
-        [InlineData(WatcherChangeTypes.All, "C:", "foo.txt")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData(WatcherChangeTypes.Changed, "C:", "foo.txt", "C:\\foo.txt")]
```

Comment:
This case is not right, `expectedFullPath` should be `{yourCurrentWorkingDirectory}\foo.txt`.

The test is passing because you are calling `Path.Join(directory, name)` before `Path.GetFullPath(directory)` and the former method doesn't treat `C:` as "your CWD in drive C", it is just concatenating both paths with a `\` in the middle.

The fix should be very simple, flip the order that the methods are called. I envision that this will now break the test cases where you pass string.Empty as directory, that is ok, i.e: it is OK that Path.GetFullPath throws for an empty path.

Reply:
Fixed in the latest commit (f28bf24)

Reply:
Hi @Jozkee. Are you happy with the changes? Can this comment be resolved?

I haven't seen any answer in a week to the above thread with moving on without caching. Should we wait longer?

Reply:
We can move on, thanks for fixing it.
Sorry for the delay, I have been heads down addressing other work items, I'm still interested in moving forward with your PR if you are as well.

Reply:
Yes, of course. I will implement your feedback soon.